arch/xtensa/core/atomic.S

```diff
@@ -393,13 +393,14 @@ atomic_cas:
 	ENTRY(48)
 .L_LoopCas:
 	l32ai a5, a2, 0
-	beq a5, a3, 1f
-	movi a2, 0
-	j 2f
+	beq a5, a3, 2f
 1:
+	movi a2, 0
+	j 3f
+2:
 	wsr a5, scompare1
 	s32c1i a4, a2, 0
-	bne a3, a5, .L_LoopCas
+	bne a4, a5, 1b
```

Comment:
Jumping back to 1 means that this will only loop once, as 1 is followed by a jump to the end?  That doesn't seem right.